service_identity/src/main/kotlin/org/thechance/service_identity/data/geteway/DataBaseGatewayImp.kt

```diff
@@ -1,85 +1,260 @@
+            filter = UserDetailsCollection::userId eq newAddressCollection.userId,
+            update = push(UserDetailsCollection::addresses, newAddressCollection.id)
+        )
+        return addressCollection.insertOne(newAddressCollection).wasAcknowledged()
+    }
 
+    override suspend fun deleteAddress(id: String): Boolean {
+        userDetailsCollection.updateOne(
+            filter = UserDetailsCollection::addresses contains ObjectId(id),
+            update = pull(UserDetailsCollection::addresses, ObjectId(id))
+        )
+        return addressCollection.updateOne(
+            filter = Filters.and(AddressCollection::id eq ObjectId(id), AddressCollection::isDeleted eq false),
+            update = setValue(AddressCollection::isDeleted, true)
+        ).isUpdatedSuccessfully()
+    }
 
-    //region userDetails
+    override suspend fun updateAddress(id: String, address: Address): Boolean {
+        return addressCollection.updateOneById(ObjectId(id), address.toCollection()).isUpdatedSuccessfully()
+    }
 
-    //endregion
+    override suspend fun getAddress(id: String): Address {
+        return addressCollection.findOne(
+            AddressCollection::id eq ObjectId(id),
+            AddressCollection::isDeleted eq false
+        )?.toEntity() ?: throw ResourceNotFoundException(NOT_FOUND)
+    }
+
+    override suspend fun getUserAddresses(userId: String): List<Address> {
+        return addressCollection.find(
+            AddressCollection::userId eq ObjectId(userId),
+            AddressCollection::isDeleted eq false
+        ).toList().toEntity()
+    }
 
+    //endregion
 
     //region Permission
-    override suspend fun getPermission(permissionId: String): Permission? {
-        return permissionCollection.findOneById(ObjectId(permissionId))?.toEntity()
-            ?: throw Exception("Wallet not found")
+    override suspend fun getPermission(permissionId: Int): Permission {
+        return permissionCollection.findOneById(permissionId)?.toEntity()
+            ?: throw ResourceNotFoundException(NOT_FOUND)
     }
 
     override suspend fun addPermission(permission: Permission): Boolean {
         return permissionCollection.insertOne(permission.toCollection()).wasAcknowledged()
-
     }
 
-    override suspend fun deletePermission(permissionId: String): Boolean {
+    override suspend fun deletePermission(permissionId: Int): Boolean {
         return permissionCollection.updateOne(
             filter = Filters.and(
-                PermissionCollection::id eq ObjectId(permissionId),
+                PermissionCollection::id eq permissionId,
                 PermissionCollection::isDeleted eq false
             ),
             update = setValue(PermissionCollection::isDeleted, true)
         ).isUpdatedSuccessfully()
     }
 
-    override suspend fun updatePermission(permissionId: String, permission: Permission): Boolean {
+    override suspend fun getListOfPermission(permissionId: Int): List<Permission> {
```

Comment:
why not getPermissions ?

Reply:
Ok, we will refractor it to the suggested name